sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobOutputStream.java

```diff
@@ -163,21 +152,16 @@ private BlobOutputStream(final BlobAsyncClient parentBlob) throws StorageExcepti
      */
     BlobOutputStream(final AppendBlobAsyncClient parentBlob, final AppendBlobAccessConditions accessCondition)
         throws StorageException {
-        this((BlobAsyncClient) parentBlob);
-        this.streamType = BlobType.APPEND_BLOB;
-
-        this.accessCondition = new BlobAccessConditions();
-        if (accessCondition != null) {
-            this.appendPositionAccessConditions = accessCondition.appendPositionAccessConditions();
-            this.accessCondition = new BlobAccessConditions().modifiedAccessConditions(accessCondition.modifiedAccessConditions()).leaseAccessConditions(accessCondition.leaseAccessConditions());
-            if (accessCondition.appendPositionAccessConditions().appendPosition() != null) {
-                this.initialBlobOffset = accessCondition.appendPositionAccessConditions().appendPosition();
-            } else {
-                this.initialBlobOffset = parentBlob.getProperties().block().value().blobSize();
-            }
-        }
-
-        this.internalWriteThreshold = (int) BlockBlobAsyncClient.BLOB_DEFAULT_UPLOAD_BLOCK_SIZE;
+        this(parentBlob, accessCondition != null ? new BlobAccessConditions().modifiedAccessConditions(
```

Comment:
This is hard to discern. I wouldn't add the constructor above.

Reply:
yeah, this was one of the most annoying classes to fix XD

Constructor sets fields based on multiple conditions.
I added the private constructor to make sure all fields are always assigned so fields can be `final` without duplicating the code in each constructor.

The other option is to add private method that set fields but that would make fields `not final`

Options here are:
1) Update code to make fields not final by adding private methods to set fields. This will simplify constructor logic
2) Add some code duplicity for assigning fields in two constructors to enhance readability for this constructor.
3) Avoid code duplicity by adding hard-to-discern calls (like it is right now)
4) Create a ticket and assign to some code owner to decide what he/she wants to do and assign a suppress rule for it
5) Any other idea is welcome 

what do you think @conniey

Reply:
This makes it hard to read. There are several ternary operators here and not clear what it's doing. Given that it's calling another overloaded constructor, you can't have any other statements before calling `this()` either. So, I would lean towards making it readable/maintainable even if that results in a tiny bit of code duplication.

Reply:
This could be a case to break this down into four classes, one which handles the interface/abstract methods, and three more which implement that for each individual use case.

Reply:
Taking number #4 after talking to Conniey.
Tiket created: https://github.com/Azure/azure-sdk-for-java/issues/5021

Rule to be supressed in this file until issue is fixed

Reply:
I made a mistake while trying to fix this PR and had it all wrong.
PR got closed after deleting branch from origin
I created a new one as this one should have been https://github.com/Azure/azure-sdk-for-java/pull/5022